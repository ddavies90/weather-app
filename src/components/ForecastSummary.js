import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastSummary.css";
import WeatherIcon from "react-icons-weather";

const ForecastSummary = (props) => {
  const { date, temperature, description, icon, onSelect } = props;
  const convertedDate = new Date(date).toLocaleDateString("default", {
    weekday: "short",
    day: "numeric",
    month: "short"
  });
  // const year = new Date(date).getFullYear().toString();
  // const dateStr = convertedDate.replace(` ${year}`, "");

  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{convertedDate}</div>
      <WeatherIcon name="owm" iconId={icon} data-testid="forecast-icon" />
      <div className="forecast-summary__temperature">
        {temperature.max}&deg;C
      </div>
      <div className="forecast-summary__description">{description}</div>
      <button
        type="button"
        data-testid="button-click"
        className="forecast-summary__btn"
        onClick={() => onSelect(date)}
      >
        More dets
      </button>
    </div>
  );
};

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    max: PropTypes.number,
    min: PropTypes.number
  }).isRequired,
  onSelect: PropTypes.func.isRequired
};

export default ForecastSummary;
