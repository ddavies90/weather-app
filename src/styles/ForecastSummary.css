.forecast-summary {
  display: flex;
  flex-direction: column;
  align-items: center;
}

div {
  padding: 0.1rem;
}
