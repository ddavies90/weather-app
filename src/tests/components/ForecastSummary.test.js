import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary />", () => {
  const validProps = {
    id: 1,
    date: 1525046400000,
    description: "Clear",
    icon: "800",
    temperature: { max: 11, min: 4 }
  };

  const renderData = (properties) => {
    return render(
      <ForecastSummary
        id={properties.id}
        date={properties.date}
        description={properties.description}
        icon={properties.icon}
        temperature={properties.temperature}
      />
    );
  };

  it("renders correctly", () => {
    const { asFragment } = renderData(validProps);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props and in the correct formats", () => {
    renderData(validProps);

    expect(screen.getByText("11Â°C")).toHaveClass(
      "forecast-summary__temperature"
    );
    expect(screen.getByText("Mon, 30 Apr")).toHaveClass(
      "forecast-summary__date"
    );
    expect(screen.getByText("Clear")).toHaveClass(
      "forecast-summary__description"
    );
    expect(screen.getByTestId("forecast-icon")).toHaveClass("wi-day-sunny");
  });
});
