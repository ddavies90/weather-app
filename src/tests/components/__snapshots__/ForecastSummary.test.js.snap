// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
    id="1"
  >
    <div
      class="forecast-summary__date"
    >
      Mon, 30 Apr
    </div>
    <i
      class="wi wi-day-sunny"
      data-testid="forecast-icon"
      name="owm"
    />
    <div
      class="forecast-summary__temperature"
    >
      11Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Clear
    </div>
  </div>
</DocumentFragment>
`;
